server.go

```diff
@@ -265,20 +265,22 @@ func (a *App) runServer(ctx context.Context) {
 			continue
 		}
 
-		switch envelope.Tag {
-		case Reply:
-			if err := a.handleFromReply(ctx, envelope.Message); err != nil {
-				log.Err(err).Msg("handle_from_reply")
+		go func() {
```

Comment:
In theory this is fine, it will work and you can immediately pull the next message immediately, hence you are not blocked on the message being processes.

But (bear with me here) we know that calls to one of the handlers can(will) take sometime, during this time there might be more messages to handle, right? and with each new message we will start a new go routine, so theoretically we can end up with thousands of Go routines.

I know this probably will not happen in practice, but you can't leave it to chances. Memory allocated by a go process is not given back to the system later. Hence, doing this can allocate a lot of memory when u have a burst of messages, but then this memory is not gonna be given to the system again.

So instead, you need to work with a pool of workers. You know exactly how many workers can be active at thes ame time, and on new messages, you give to a free worker. this is very easy to implement in go actually.

```go
func worker(in <- chan Envlope) {
   for envelope := range in { 
      // do switch on tag thingy here
     // never break out of the loop. 
   }
} 

// start the workers
ch := make(chan Envelope);
for i:=0; i<10; i++ {
    go worker(ch)
}

// now feeding the workers
for {
  // pull for new message
  ch <- envelope // the first free worker will receive it
}
``` 

This is just some pseudo code, needs improvements.